ddtrace/internal/writer.py

```diff
@@ -458,16 +459,17 @@ def flush_queue(self, raise_exc=False):
                 self._metrics_dist("http.errors", tags=["type:err"])
                 self._metrics_dist("http.dropped.bytes", len(encoded))
                 self._metrics_dist("http.dropped.traces", n_traces)
+                log_msg = "failed to send, dropping %d traces to intake at %s after %d retries (%s)" % (
+                    n_traces,
+                    self._intake_endpoint,
+                    e.last_attempt.attempt_number,
+                    e.last_attempt.exception(),
+                )
+                telemetry_writer.add_error(trace_telemetry.AGENT_ERROR_CODE, log_msg)
                 if raise_exc:
                     e.reraise()
                 else:
-                    log.error(
```

Comment:
patch the DDLogger (internal.logger.DDLogger) instead to add telemetry writer errors

```
    def error(self, *args, **kwargs):
        telemetry_writer.add_error()
        return super(DDLogger, self).error(*args, **kwargs)
```